tests/integration/conftest.py

```diff
@@ -206,6 +213,22 @@ def patch_salt_globals_folder_state(vmware_conf):
     setattr(folder_state, "__pillar__", vmware_conf)
 
 
+@pytest.fixture
+def patch_salt_globals_datacenter(vmware_conf):
+    """
+    Patch __opts__ and __pillar__
+    """
+
+    setattr(
+        datacenter_mod,
+        "__opts__",
+        {
+            "test": False,
+        },
+    )
+    setattr(datacenter_mod, "__pillar__", vmware_conf)
```

Comment:
100% on the patch, I'm not sure on the yield because `patch_salt_globals` is run for each test; given the speed at which it executes and the cost of a test accidentally forgetting to undo any alterations I think it should stay that way.

Reply:
[Duplicated from below](https://github.com/saltstack/salt-ext-modules-vmware/pull/216/files#r809565866):

---------------

A better way would be for `vmware_datacenter` to have the `patch_salt_globals_datacenter` fixture in its argument list;

```python
@pytest.fixture(scope="function")
def vmware_datacenter(patch_salt_globals, service_instance):
    """
    Return a vmware_datacenter during start of a test and tear it down once the test ends
    """
    dc_name = str(uuid.uuid4())
    dc = datacenter_mod.create(name=dc_name, service_instance=service_instance)
    yield dc_name
    datacenter_mod.delete(name=dc_name, service_instance=service_instance)
```

would be 


```python
@pytest.fixture
def patch_salt_globals_datacenter(vmware_conf, patch_salt_globals):
    setattr(
        datacenter_mod,
        "__opts__",
        {
            "test": False,
        },
    )
    setattr(datacenter_mod, "__pillar__", vmware_conf)

@pytest.fixture(scope="function")
def vmware_datacenter(patch_salt_globals_datacenter, service_instance):
    """
    Return a vmware_datacenter during start of a test and tear it down once the test ends
    """
    dc_name = str(uuid.uuid4())
    dc = datacenter_mod.create(name=dc_name, service_instance=service_instance)
    yield dc_name
    datacenter_mod.delete(name=dc_name, service_instance=service_instance)
```

--------

With respect to `patch` and `yield`, it's much better than `setattr` because that way tests that *don't* rely on `patch_salt_globals.*` fixtures won't accidentally use things set by other tests; In general I really dislike it when tests are dependent on other tests to run first.

Reply:
I will fix this in a new pr.